bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -97,6 +97,15 @@ def run_inner_plan(scope, plan_name, targets, args = {})
       param_acc[param.name] = extract_parameter_types(param.type_expr)&.flatten
```

Comment:
extract_parameter_types only gets the rightmost parameter type, so for example `extract_parameter_types(Optional[TargetSpec])` will return just `TargetSpec`. This is fine when we're checking whether to populate a `$targets` since TargetSpec will always be the rightmost type, but for checking `Sensitive` it means the `Sensitive[String]` is a valid type but gets missed by this type checker.

I think the best thing to do would be to have extract_parameter_types traverse `l_expr` recursively in addition to `keys`, `types`, etc. then check whether any of  `param_types[param]` (which should be an array) is sensitive.